pkg/backup/backup_compaction.go

```diff
@@ -653,3 +703,71 @@ func processProgress(
 	}
 	return nil
 }
+
+type compactionChain struct {
+	// restoreChain is the linear chain of backups up to the end time required
+	// for a restore.
+	restoreChain   []backuppb.BackupManifest
+	chainToCompact []backuppb.BackupManifest
+	start, end     hlc.Timestamp
```

Comment:
nit: clarify that `end` represents the end time of the last backup to be compacted and start is the start time of the first backup to be compacted